extensions/transform/org.eclipse.smarthome.transform.regex/src/main/java/org/eclipse/smarthome/transform/regex/internal/RegExTransformationService.java

```diff
@@ -87,7 +91,11 @@ public String transform(String regExpression, String source) throws Transformati
             }
         }
 
-        return result;
+        if (result != null) {
+            return result;
+        } else {
+            return UnDefType.NULL.toFullString();
```

Comment:
This line is to be conform with https://github.com/eclipse/smarthome/blob/master/extensions/transform/org.eclipse.smarthome.transform.jsonpath/src/main/java/org/eclipse/smarthome/transform/jsonpath/internal/JSonPathTransformationService.java#L60

because this method should not return `null`.